codex-rs/app-server-protocol/src/export.rs

```diff
@@ -176,22 +146,82 @@ pub fn generate_json(out_dir: &Path) -> Result<()> {
         "ServerRequest",
     ];
 
-    for (name, schema) in bundle {
-        let mut schema_value = serde_json::to_value(schema)?;
-        annotate_schema(&mut schema_value, Some(name.as_str()));
+    // Merge all generated per-type JSON files (including the envelopes above)
+    // into a single definitions bundle. Also recognize a v2 namespace so that
+    // types like RateLimitSnapshot can co-exist between legacy and v2.
+    let json_files = json_files_in_recursive(out_dir)?;
```

Comment:
This PR seems to add quite a lot of new code to this file: can it be broken out?

For example, there seems to be a lot of JSON schema stuff, which feels like it could be separated?